.github/workflows/test.yml

```diff
@@ -1,27 +1,30 @@
 name: Test
 
-on: [push, pull_request]
+on: [pull_request]
 
 # Cancel jobs and just run the last one
 concurrency:
-  group: ${{ github.head_ref }}-unit-test
+  group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
   unit_test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: subosito/flutter-action@v2
-        with:
-          flutter-version: '3.0.5'
-          channel: 'stable'
-          cache: true
-          cache-key: flutter-3 # optional, change this to force refresh cache
-          cache-path: ${{ runner.tool_cache }}/flutter # optional, change this to specify the cache path
-
-      - name: Install dependencies
-        run: flutter pub get
+    - uses: actions/checkout@v3
+    - uses: ./.github/actions/setup-flutter
+    - name: Run tests
+      run: flutter test
+  
+  test_publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./.github/actions/setup-flutter
 
-      - name: Run tests
-        run: flutter test
\ No newline at end of file
+    - name: dry-run publishing to feel confident deployment will work after merging PR
+      uses: k-paxian/dart-package-publisher@v1.6
+      with:
+        credentialJson: ${{ secrets.CREDENTIAL_JSON }}
+        dryRunOnly: true 
+        flutter: true
```

Comment:
This code the way that it is will do nothing, unfortunately. 

[If you view the output in the CI](https://github.com/customerio/customerio-flutter/actions/runs/5656432153/job/15323447896#step:5:187), you will notice that dry-run exits early because the version number has not been changed yet. 

This is the reasoning why I was experimenting with [this hack](https://github.com/customerio/customerio-flutter/pull/60/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R25-R34) of incrementing the version of the package and modifying the changelog to trick flutter into thinking we were introducing a new version. Which, then will continue running the dry-run deployment instead of exiting early. 

I hope there is a better solution then what I came up with. However, it's not that complicated of a solution so if we have to use it, it's still an option.

Reply:
Good point, maybe we can do this before attempting to release. Let me think more about how this can be done.